### DISCUSSION TITLE
Log "Context" does not show up in Dashboards

### DISCUSSION THREAD

**Discussion by itsek** [grafana/grafana#58408] [Observability]
**What happened**:
Using Loki as Data Source in Grafana 8.5.6, Logs are Filtered using a filter operator.
https://grafana.com/blog/2019/08/14/new-in-grafana-v6.3-introducing-lokis-log-row-context-viewer/

This Blog Post Explained that in this configuration I should have a "Context" Button to see Logs above and below
There is no Context Button

[![Screenshot-2022-11-03-154900.png](https://i.postimg.cc/vZPmP2CD/Screenshot-2022-11-03-154900.png)](https://postimg.cc/YvWk0Rrw)

**What you expected to happen**:
Context Button appearence

**How to reproduce it (as minimally and precisely as possible)**:
Query some Loki Logs, use a filter operator, check for Button

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.5.6
- Data source type & version: Loki
- User OS & Browser: Windows / Firefox

**Comment by itsek** [comment]
Found out that it only works in "Explore" but not in a Dashboard - Please convert this to a feature request: "Context in Dashboards"

**Comment by usmangt** [comment]
Hi @itsek,

Thanks for opening this issue.

I will move it into our Discussions section so that it get under the Feature request section.

**Comment by vasuarv** [comment]
This will be very useful. Currently, the dashboard which pulls in the log entries is quite limited.

**Comment by SunSparc** [comment]
I would love this as well. Some of my users do not have permission to use the Explorer, but they do view logs. There are many feature of the Explorer tool that I would like to be able to let my "viewers" have access to. The "Show Context" feature would be a great start.

**Reply by stevendungan** [reply]
@SunSparc would the [viewers_can_edit](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#viewers_can_edit) flag work for your use case here? If not, curious about why, and your org structure in general (and how you would/would not choose to assign Editor roles)


Also, curious about the other use cases in Explore non-editors would be interested in.

**Reply by SunSparc** [reply]
@stevendungan, the `viewers_can_edit` setting could be useful for us. I will have to evaluate that. I have been using Grafana for a few years now, but I have only recently started introducing other employees into the environment. I have no played much with the permissions and have opted to be more strict to start off and then loosen permissions as needed. One reason to not assign "Editor" roles is the concern that people would break/delete dashboards/panels inadvertently because of ignorance. As we bring people up to speed on things, I think we will be more likely to hand out "Editor" roles, especially per dashboard.

Concerning other features of the Explorer, in some cases I would like to just make log dashboards that are duplicates of the Explorer, except with pre-populated queries/settings. The viewing controls (time, unique, wrap, prettify, dedupe, newest/oldest first) are very handy to have on the log screen without needing to open the panel editor. It would be nice to have the option to include those on the panel view screen.

**Comment by stevendungan** [comment]
@itsek @vasuarv could you tell me a bit about how you are using Dashboard for logs, and where using Explore doesn't work as well?

**Comment by itsek** [comment]
Hi,

We provide a "pre configured" Dashboard for let's say the Application Manager, who does not know LogQL (has no time learning it)
So the Idea is to deploy a dashboard which is "ready to use" with pre-populated filtering/formatting/relabeling etc. 
The User sees a "all errors" or "all warning" or "top 10 errors from app-x" and then can just browse the stuff

Because now we say "hey, copy that very long logQL line into the explore mode and then you can do context search :)

**Reply by stevendungan** [reply]
thank you! very helpful insight.

**Reply by stevendungan** [reply]
if you're able to do it without providing sensitive information; any chance you could share a screenshot of one of the pre-configured dashboards you use? completely understand if you cannot do this.

**Reply by itsek** [reply]
@stevendungan Oh well, all blurred out in the screenshot, but the Idea is as outlined above: provide pre-configured Panels for Users, so they can quickly find what they need without knowledge of the underlying LogQL inner-workings.

![2023-01-30_083049](https://user-images.githubusercontent.com/76947368/215415789-a3cd5f93-406b-46c5-ac67-3a91ae503a02.png)

**Comment by ivanahuckova** [comment]
For additional discussion, please use https://github.com/grafana/grafana/issues/20166
